tests/modules/commerce_redirect_test/src/Form/RedirectTestForm.php

```diff
@@ -0,0 +1,35 @@
+<?php
+
+namespace Drupal\commerce_redirect_test\Form;
+
+use Drupal\commerce\Response\NeedsRedirectException;
+use Drupal\Core\Form\FormBase;
+use Drupal\Core\Form\FormStateInterface;
+
+/**
+ * Tests throwing a redirect inside of a form.
+ */
+class RedirectTestForm extends FormBase {
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getFormId() {
+    return 'commerce_redirect_test_form';
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function buildForm(array $form, FormStateInterface $form_state) {
+    throw new NeedsRedirectException('https://www.drupal.org');
```

Comment:
Thanks @timplunkett! I updated our test to use that, and mention we do not need to use our method